app/interactors/outdated_prolem_clearer.rb

```diff
@@ -0,0 +1,32 @@
+require 'problem_destroy'
+
+class OutdatedProblemClearer
+
+  ##
+  # Clear all problem not present for more than one week.
+  #
+  def execute
+    nb_problem_outdated.tap { |nb|
+      if nb > 0
+        criteria.each do |problem|
+          ProblemDestroy.new(problem).execute
+        end
+        repair_database
+      end
+    }
+  end
```

Comment:
I've added a spec similar to the `ResolvedProblemClearer` spec. Did not find the time to test this locally, let's see 🤠